backend/app/api/endpoints/share.py

```
@@ -0,0 +1,542 @@
+    except ValueError as e:
+        raise HTTPException(status_code=400, detail=str(e)) from e
+
+
+@router.get(
+    "/public/knowledge/redirect",
+    response_model=Dict[str, str],
+    summary="Get share token for KB redirect",
+)
+def get_kb_share_token(
+    kb_id: int,
+    db: Session = Depends(get_db),
```

Ruff (0.14.14)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)